File: nc.productmanager.Data/Models/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace nc.productmanager.Data.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int ProductCategoryId { get; set; }
        [ForeignKey(nameof(ProductCategoryId))]
        public ProductCategory ProductCategory { get; set; }
    }
}

File: nc.productmanager.Client/src/app/models/Product.ts
import { GetProductCategory } from "./ProductCategory";

export interface GetProduct {
  id: number;
  name: string;
  description: string;
  price: number;
  category: GetProductCategory;
}

export interface Product {
  name: string;
  description: string;
  price: number;
  productCategoryId: number;
}
